src/api-next/api-next.ts

```diff
@@ -2041,22 +2044,48 @@ export class IBApiNext {
     order: Order,
     orderState: OrderState
   ): void => {
+    // console.log('onOpenOrder', orderId, order.permId);
     subscriptions.forEach((sub) => {
       const allOrders = sub.lastAllValue ?? [];
-      allOrders.push({ orderId, contract, order, orderState });
-      sub.next({
-        all: allOrders,
-      });
+      const changeOrderIndex = allOrders.findIndex(
+        (p) => p.order.permId == order.permId
+      );
+      if (changeOrderIndex === -1) {
+        // new open order - add it
+        const addedOrder: OpenOrder = { orderId, contract, order, orderState, orderStatus: undefined };
+        allOrders.push(addedOrder);
+        sub.next({
+          all: allOrders,
+          added: [addedOrder],
+        });
+      } else {
+        // update
+        const updatedOrder: OpenOrder = allOrders[changeOrderIndex];
+        updatedOrder.order = order;
+        updatedOrder.orderState = orderState;
+        if (updatedOrder.orderStatus !== undefined) {
+          // synchronize orderStatus if exists
+          updatedOrder.orderStatus.clientId = order.clientId;
+          updatedOrder.orderStatus.permId = order.permId;
+          updatedOrder.orderStatus.parentId = order.parentId;
+          updatedOrder.orderStatus.status = orderState.status;
+        }
+        sub.next({
+          all: allOrders,
+          changed: [updatedOrder],
+        });
+      }
     });
   };
 
   /**
    *  Ends the subscrition once all openOrders are recieved
-   *  @param subscriptions
+   *  @param subscriptions: listeners
    */
   private readonly onOpenOrderEnd = (
     subscriptions: Map<number, IBApiNextSubscription<OpenOrder[]>>
   ): void => {
+    // console.log('onOpenOrderEnd');
```

Comment:
you can keep the logger if you set it to, also need to import logger for that


```suggestion
    logger.debug('onOpenOrderEnd');
```

`logger.debug(`

Reply:
It can be replaced with a `logger.debug`

Next time easier to see what's up. Also while feeding in the Logger level from consuming the app.